modules/light-clients/08-wasm/keeper/keeper.go

```diff
@@ -134,3 +145,50 @@ func (k Keeper) storeWasmCode(ctx sdk.Context, code []byte) ([]byte, error) {
 
 	return codeHash, nil
 }
+
+func (k Keeper) migrateContractCode(ctx sdk.Context, clientID string, newCodeHash, migrateMsg []byte) error {
+	wasmClientState, err := k.GetWasmClientState(ctx, clientID)
+	if err != nil {
+		return errorsmod.Wrap(err, "failed to retrieve wasm client state")
+	}
+	oldCodeHash := wasmClientState.CodeHash
+
+	clientStore := k.clientKeeper.ClientStore(ctx, clientID)
+
+	err = wasmClientState.MigrateContract(ctx, k.cdc, clientStore, clientID, newCodeHash, migrateMsg)
+	if err != nil {
+		return errorsmod.Wrap(err, "contract migration failed")
+	}
+
+	// client state may be updated by the contract migration
+	wasmClientState, err = k.GetWasmClientState(ctx, clientID)
```

Comment:
would be nice to have cov for the `err` case

Reply:
done